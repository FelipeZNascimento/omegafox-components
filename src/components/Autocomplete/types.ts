export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
