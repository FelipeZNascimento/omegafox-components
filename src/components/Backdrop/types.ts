export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
