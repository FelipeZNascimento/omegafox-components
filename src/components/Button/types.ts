export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
