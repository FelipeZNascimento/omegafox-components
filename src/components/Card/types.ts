export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
