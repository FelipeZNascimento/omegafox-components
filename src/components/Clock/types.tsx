import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
