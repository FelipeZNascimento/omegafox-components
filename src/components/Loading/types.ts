export interface LoadingProps {
  image: string;
  overlay?: boolean;
  shadowed?: boolean;
  size?: 'small' | 'regular' | 'big';
  style?: 'spin' | 'headbutt';
  text?: string;
}
