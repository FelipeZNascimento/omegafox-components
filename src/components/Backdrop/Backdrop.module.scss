@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include bg-rgba-colored($color-black, 0);
  transition: background-color $transition-time;
  position: fixed;
  left: 0;
  top: 0;
  height: 100vh;
  width: 100vw;
  z-index: $always-on-top + 1;

  &Left {
    @include flex-aligned(flex-start, center);
  }

  &Center {
    @include flex-aligned(center, center);
  }

  &Right {
    @include flex-aligned(flex-end, center);
  }

  &Open {
    @include bg-rgba-colored($color-black, 0.4);
    transition: background-color $transition-time;
  }

  // &Closed {
  // }
}
