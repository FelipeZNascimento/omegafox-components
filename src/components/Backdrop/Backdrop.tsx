import React, { useEffect, useState } from 'react';

import classNames from 'classnames';
import { IBackdropProps } from './types';
import styles from './Backdrop.module.scss';

export const Backdrop = ({
  align = 'center',
  children,
  isOpen
}: IBackdropProps) => {
  const [isContentOpen, setIsContentOpen] = useState<boolean>(false);

  useEffect(() => {
    if (isOpen && !isContentOpen) {
      setTimeout(function () {
        setIsContentOpen(true);
      }, 200);
    } else {
      setTimeout(function () {
        setIsContentOpen(false);
      }, 200);
    }
  }, [isContentOpen, isOpen]);

  const backgroundClass = classNames(styles.container, {
    [styles.containerLeft]: align === 'left',
    [styles.containerCenter]: align === 'center',
    [styles.containerRight]: align === 'right',
    [styles.containerOpen]: isOpen && isContentOpen
  });

  return (
    <div className={backgroundClass}>{isOpen && isContentOpen && children}</div>
  );
};
