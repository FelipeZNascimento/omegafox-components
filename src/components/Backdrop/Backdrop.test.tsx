import React from 'react';
import { render } from '@testing-library/react';
import { Backdrop } from './Backdrop';
import { IBackdropProps } from './types';

describe('<Backdrop />', () => {
  const renderComponent = ({ align, children, isOpen }: IBackdropProps) =>
    render(
      <Backdrop align={align} isOpen={isOpen}>
        {children}
      </Backdrop>
    );

  it('should render', () => {
    const { container } = renderComponent({
      align: 'center',
      children: <p>anything</p>,
      isOpen: true
    });

    expect(container).toMatchSnapshot();
  });
});
