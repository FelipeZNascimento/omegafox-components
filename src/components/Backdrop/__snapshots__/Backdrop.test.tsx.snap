// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> should render 1`] = `
<div>
  <div
    class="container containerCenter"
  />
</div>
`;
